docarray/array/mixins/embed.py

```diff
@@ -11,29 +17,40 @@ class EmbedMixin:
     def embed(
         self: 'T',
         embed_model: 'AnyDNN',
+        collate_fn: Optional['CollateFnType'] = None,
         device: str = 'cpu',
         batch_size: int = 256,
         to_numpy: bool = False,
     ) -> 'T':
         """Fill :attr:`.embedding` of Documents inplace by using `embed_model`
 
         :param embed_model: the embedding model written in Keras/Pytorch/Paddle
+        :param collate_fn: create a mini-batch of Input(s) from the given `DocumentArray`.  Default built-in collate_fn
+                is to use the `tensors` of the documents.
         :param device: the computational device for `embed_model`, can be either
             `cpu` or `cuda`.
         :param batch_size: number of Documents in a batch for embedding
         :param to_numpy: if to store embeddings back to Document in ``numpy.ndarray`` or original framework format.
         :return: itself after modified.
         """
 
+        if collate_fn is None:
+
+            def default_collate_fn(da: 'DocumentArray'):
+                return da.tensors
```

Comment:
i think the default collate function should convert `tensors` property to `tensor` objects, such as pytorch tensor.